Chimera master bc4 feature 6530

rommstein opened the pull request.

rommstein committed:
BJS2-6530

rommstein committed:
BJS2-6530
=)))

eRqa commented on src/main/java/faang/school/postservice/dto/comment/CommentDto.java:
```diff
@@ -0,0 +1,19 @@
+@AllArgsConstructor
+@Builder
+public class CommentDto {
+    private Long id;
```
Здесь нужно добавить валидацию обязательных полей аннотациями. У тебя уже есть задача по этой теме https://faang-school.atlassian.net/jira/software/c/projects/BJS2/boards/45?selectedIssue=BJS2-6726
В новых ПРах теперь обязательное требование делать валидацию ДТОшек имеено аннотациями

eRqa commented on src/main/java/faang/school/postservice/service/comment/CommentService.java:
```diff
@@ -0,0 +1,81 @@
+    public CommentDto createComment(CommentDto commentDto) {
+        validateCommentDto(commentDto);
+        Comment comment = commentMapper.fromDto(commentDto);
+        comment.setCreatedAt(LocalDateTime.now());
```
Это можно убрать так как в репозитории по умолчанию этому полю (и другим датам) будет проставлено LocalDateTime.now()

eRqa commented on src/main/java/faang/school/postservice/service/comment/CommentService.java:
```diff
@@ -0,0 +1,81 @@
+    }
+
+    public CommentDto updateComment(long id, CommentDto commentDto) {
+        validateCommentDto(commentDto);
```
Это нужно переделать на аннотации

eRqa commented on src/main/java/faang/school/postservice/mapper/post/CommentMapper.java:
```diff
@@ -0,0 +1,12 @@
+import org.mapstruct.Mapper;
+
+@Mapper(componentModel = "spring")
+public interface CommentMapper {
```
На этот класс нужно написать тесты. Когда напишешь, увидишь что есть ошибки, не все поля смапплены

rommstein committed:
BJS2-6530
Убрал LocalDateTime.now() из сервиса
Добавил аннотацию в CommentMapper и написал тесты

eRqa commented on src/main/java/faang/school/postservice/controller/comment/CommentController.java:
```diff
@@ -0,0 +1,59 @@
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/comments")
+public class CommentController {
```
Я короче посмотрел, и понял что здесь не совсем правильно построено API
Здесь теряются лайки, в Comment они есть, а CommentDTO их нет. То есть фронт никогда не будет знать сколько лайков у коммента, и никогда не сможет обновить количество. Нужно это доработать)
В методе create лайки не нужны, так как когда коммент создается лайков 0, а вот при update и get нужны
Если не знаешь как сделать, можем обсудить

eRqa commented on src/test/java/faang/school/postservice/mapper/post/CommentMapperTest.java:
```diff
@@ -0,0 +1,75 @@
+@ExtendWith(MockitoExtension.class)
+public class CommentMapperTest {
+
+    @Mock
```
А зачем здесь мока на пост? Это же не зависимость)

rommstein committed:
BJS2-6530
Исправил CommentMapperTest
